overlord/servicestate/quota_control.go

```diff
@@ -58,7 +67,59 @@ func EnsureQuotaUsability() (restore func()) {
 	}
 }
 
+func checkMemoryCGroupEnabled() {
+	memoryCGroupError = memoryCGroupEnabled()
+}
+
+// added to enable path update for testing purposes
+func SetCGroupsFilePath(path string) {
+	CGroupsFilePath = path
+	checkMemoryCGroupEnabled()
+}
+
+// since the control groups can be enabled/disabled without the kernel config the only
+// way to identify the status of memory control groups is via /proc/cgroups
+// "cat /proc/cgroups | grep memory" returns the active status of memory control group
+// and the 3rd parameter is the status
+// 0 => false => disabled
+// 1 => true => enabled
+func memoryCGroupEnabled() error {
+	cgroupsFile, err := os.Open(CGroupsFilePath)
+	if err != nil {
+		return fmt.Errorf("cannot open cgroups file: %v", err)
+	}
+	defer cgroupsFile.Close()
+	scanner := bufio.NewScanner(cgroupsFile)
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "memory\t") {
+			memoryCgroupValues := strings.Fields(line)
+			if len(memoryCgroupValues) >= 4 { // assuming any increase in size will lead to values added to the end
```

Comment:
minor: I'd invert the condition and return the "cannot parse memory control group configuration" error here.

Reply:
This was an area of discussion in the previous iteration of this PR, we decided to go with the assumption that any fields added will have backwards compatibility and not mess up the first 4. Up for further discussion if it feels weird as it's definitely not a guarantee.

Reply:
I agree with the assumption, I was joint pointing out that instead of
```go
if len(memoryCgroupValues) >= 4 {
    ...do stuff...
}
return fmt.Errorf("cannot parse memory control group configuration")
```
you could do
```go
if len(memoryCgroupValues) < 4 {
    return fmt.Errorf("cannot parse memory control group configuration")
}
...do stuff...
```
which seems more readable to me.